coral/src/app/features/approvals/acls/AclApprovals.test.tsx

```diff
@@ -0,0 +1,207 @@
+    username: "amathieu",
+    requesttime: "2023-01-10T13:19:10.757+00:00",
+    requesttimestring: "10-Jan-2023 13:19:10",
+    aclstatus: "created",
+    approver: undefined,
+    approvingtime: undefined,
+    aclType: "Producer",
+    aclResourceType: undefined,
+    currentPage: "1",
+    otherParams: undefined,
+    totalNoPages: "1",
+    allPageNos: ["1", ">", ">>"],
+    approvingTeamDetails:
+      "Team : Ospo, Users : muralibasani,josepprat,samulisuortti,mirjamaulbach,smustafa,aindriul,",
+  },
+];
+
+const mockGetAclRequestsForApproverResponse = transformAclRequestApiResponse(
+  mockedAclRequestsForApproverApiResponse
+);
+
+describe("AclApprovals", () => {
+  beforeAll(() => {
+    mockIntersectionObserver();
+  });
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  describe("Skeleton", () => {
+    beforeAll(() => {
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      //@ts-ignore
+      useQuerySpy.mockReturnValue({ data: [], isLoading: true });
+
+      customRender(<AclApprovals />, {
+        queryClient: true,
+        memoryRouter: true,
+      });
+    });
+
+    afterAll(() => {
+      cleanup();
+      useQuerySpy.mockRestore();
+    });
+
+    it("shows a skeleton table", () => {
+      const skeleton = screen.getByTestId("skeleton-table");
+
+      expect(skeleton).toBeVisible();
+    });
+  });
+
+  describe("DataTable", () => {
+    beforeAll(async () => {
+      mockGetAclRequestsForApprover.mockResolvedValue(
+        mockGetAclRequestsForApproverResponse
+      );
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      //@ts-ignore
+      customRender(<AclApprovals />, {
+        queryClient: true,
+        memoryRouter: true,
+      });
+
+      await waitForElementToBeRemoved(screen.getByTestId("skeleton-table"));
+    });
+
+    afterAll(() => {
+      cleanup();
+    });
+
+    it("shows one header row and two data rows", async () => {
+      const rows = screen.getAllByRole("row");
+
+      expect(rows).toHaveLength(3);
+      expect(rows[0]).toHaveTextContent("Principals/Usernames");
+      expect(rows[1]).toHaveTextContent("mbasani");
+      expect(rows[2]).toHaveTextContent("User:*");
+    });
+
+    it("renders (prefixed) in Topic cell when appropriate", async () => {
+      const prefixedCells = screen.getAllByText("(prefixed)");
+      const notPrefixedCells = screen.getAllByText("aivtopic1");
+
+      expect(prefixedCells).toHaveLength(1);
+      expect(notPrefixedCells).toHaveLength(1);
+    });
+
+    it("renders all values for cells who can have multiple values", async () => {
+      const cells = screen.getAllByRole("cell");
+
+      expect(
+        cells.filter((cell) => {
+          return cell.textContent === "mbasanimaulbach";
+        })
+      ).toHaveLength(1);
+      expect(
+        cells.filter((cell) => cell.textContent === "3.3.3.323.3.3.33")
+      ).toHaveLength(1);
+    });
+  });
+
+  describe("Pagination", () => {
```

Comment:
minor nitpick: I would name that test something like "handles paginated responses". The pattern `describe('Pagination')` is used for the outermost describe and indicates that these are the tests for the `Pagination` component.

Reply:
[bb0ac03](https://github.com/aiven/klaw/pull/581/commits/bb0ac03fe5cf838801926500cb5a0fb10433ac26)